kaizhilam/ProjectCreature
# Adding animations, player wields weapons now

* Integrating Stefans animation code
	* Player will play jump animation on spacebar hit
	* Player will now roll on Q press (needed to change from E because E was pickup
	* Player will stop Run animation when still
	* Player will play slash animation on left click
* Animation code went inside new AnimationManager.cs script inside Manager folder as a singleton class
* PlayerStat.cs renamed to Player.cs
* Stefans weapon collision code went inside Player.cs
* InputController can now broadcast when player has stopped moving (used to stop run animation)
* InputController once again has involvement in moving the player
* Added player animations to project
* Player->Idle in hierarchy now a prefab so I could attach weapon (pls fix?)
* Added AnimationManager to GameController Prefab
* Removing SoundController prefab as SoundManager script is now held inside GameController prefab
* Player is now 1.5x her size

--How to test this pull request--
* Q should make player dodge
* Running should play run animation
* Left click should play slash animation
* Being still should stop run animation
* Player should now wield weapon in right hand
![wield](https://user-images.githubusercontent.com/19967756/57385384-b3dc5b80-71f5-11e9-87b5-e18d5aa2f49a.PNG)

Known bugs / Future Improvements

- [x] Players weapon can obscure camera and make it zoom right in
- [x] Player shouldn't be able to spam slash, jump or dodge animations
- [ ] There should be no prefabs inside player prefab
